// component/AppBadgePoint.js
import React from 'react'

import Badge from '@material-ui/core/Badge'
import { brown, red, deepOrange } from '@material-ui/core/colors'
import Button from '@material-ui/core/Button'
import { withStyles, createMuiTheme } from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#ffffff',
    },
  },
});

const StyledBadge = withStyles((theme) => ({
  badge: {
    position: 'absolute',
    top: '98%',
    left: '35%',
    width: '2em',
    height: '1em',
    backgroundColor: red[500],
    color: red[500],
    boxShadow: `0 0 0 2px ${theme.palette.background.paper}`,
    '&::after': {
      position: 'absolute',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
      borderRadius: '50%',
      animation: '$ripple 1.2s infinite ease-in-out',
      border: '1px solid currentColor',
      content: '""',
    },
  },
  '@keyframes ripple': {
    '0%': {
      transform: 'scale(.8)',
      opacity: 1,
    },
    '100%': {
      transform: 'scale(2.4)',
      opacity: 0,
    },
  },
}))(Badge);


export const BadgePoint = (props)=> {
  return (
    <ThemeProvider theme={theme}>
        <StyledBadge
          overlap="circle"
          variant="dot"
        >
          <Button
                  variant="outlined"
                  onClick={props.addToBasket}
                  style={{
                          color: brown[500], 
                          fontWeight: 'bolder', 
                          fontSize: '1.2em', 
                          borderRadius: '.7em',
                          border: `1px solid ${brown[800]}`, 
                          background: deepOrange[50],
                        }}
          >
              Add to cart
          </Button>

        </StyledBadge>
    </ThemeProvider>
  );
}
