// component/AppDrawer.js
import React from 'react'
import clsx from 'clsx'
import { makeStyles } from '@material-ui/core/styles'
import Drawer from '@material-ui/core/Drawer'
import IconButton from '@material-ui/core/IconButton'
import List from '@material-ui/core/List'
import Grid from '@material-ui/core/Grid'
import Divider from '@material-ui/core/Divider'
import MenuIcon from '@material-ui/icons/Menu'
import Typography from '@material-ui/core/Typography'

import Sidebar from 'components/sidebar'



const useStyles = makeStyles({
  list: {
    width: 300,
    height: '100%',
    background: 'linear-gradient(0deg, rgba(62,39,35,1) 0%, rgba(239,235,233,1) 5%, rgba(239,235,233,1) 95%, rgba(62,39,35,1) 100%)',
  },
  fullList: {
    width: 'auto',
  },
});

export const AppDrawer = ()=> {
  const classes = useStyles();
  const [state, setState] = React.useState({
    left: false
  });

  const toggleDrawer = (anchor, open) => (event) => {
    if (event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
      return;
    }
    setState({ ...state, [anchor]: open });
  };

  const list = (anchor) => (
    <div
      className={clsx(classes.list, {
        [classes.fullList]: anchor === 'top' || anchor === 'bottom',
      })}
      role="presentation"
      onClick={toggleDrawer(anchor, true)}
      onKeyDown={toggleDrawer(anchor, true)}
    >
      <Grid container direction="column" justify="space-between" alignItems="center" style={{height: '100%'}}>
        <Divider />
          <List >
            <Sidebar />
          </List>
        <Divider />
      </Grid>
    </div>
  );

  return (
    <Typography component="div" style={{opacity: 0.7}}>
      {['left'].map((anchor) => (

        <React.Fragment key={anchor}>

          <IconButton onClick={toggleDrawer(anchor, true)} color="inherit">
            {<MenuIcon fontSize='large'/>}
          </IconButton>

          <Drawer anchor={anchor} open={state[anchor]} onClose={toggleDrawer(anchor, false)} >
            {list(anchor)}
          </Drawer>

        </React.Fragment>
      ))}
    </Typography>
  );
}
