// components/basketCart/index.js
import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

import Badge from '@material-ui/core/Badge'
import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import { ThemeProvider } from '@material-ui/styles'
import { green, brown, red, deepOrange } from '@material-ui/core/colors'
import ShoppingCartIcon from '@material-ui/icons/ShoppingCartOutlined'
import MonetizationOnOutlinedIcon from '@material-ui/icons/MonetizationOnOutlined'
import { withStyles, makeStyles, createMuiTheme } from '@material-ui/core/styles'

import {
  getTotalBasketCount,
  getTotalBasketPrice
} from 'selectors'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#ffffff',
    },
    secondary: {
      main: green['A200'],
    },
  },
});

const useStyles = makeStyles((theme) => ({
  basket: {
    borderRadius: '.7em',
    paddingLeft: '.5em', 
    paddingRight: '4em',
    border: `1px solid ${brown[800]}`,
    background: deepOrange[50],
    '&:hover': {
      backgroundColor: '#fddbd2',
    },
  },
}));

const StyledBadgeLeft = withStyles((theme) => ({
  badge: {
    right: '-.4em',
    bottom: 17,
    color: red[500],
    fontWeight: 'bolder',
    fontSize: '1.2em',
    border: `1px solid ${brown[200]}`,
    padding: '0 4px',
  },
}))(Badge);

const StyledBadgeRight = withStyles((theme) => ({
  badge: {
    right: '-1.4em',
    bottom: 17,
    color: red[500],
    fontWeight: 'bolder',
    fontSize: '1.2em',
    border: `1px solid ${brown[200]}`,
    padding: '0 4px',
  },
}))(Badge);



const BasketCart = ({totalBasketCount, totalPrice})=> {
  const classes = useStyles();

  return(
    <ThemeProvider theme={theme}>
      <Button variant="outlined" aria-label="cart" color='inherit' className={classes.basket} >
          <Link
            to='/basket'
            id='dLabel'
            style={{width: '100%'}}
          >
            <Grid container direction="row" justify="center" alignItems="center">

              <StyledBadgeLeft badgeContent={totalBasketCount} color='primary' anchorOrigin={{
                  vertical: 'bottom',
                  horizontal: 'right',
              }}>
                <ShoppingCartIcon fontSize='large' style={{color: brown['A400']}} />
              </StyledBadgeLeft>

              <StyledBadgeRight badgeContent={totalPrice} max={999999999} color="primary" anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'right',
              }}>
                <MonetizationOnOutlinedIcon fontSize='large' style={{color: brown['A400'], marginLeft: '.6em'}} />
              </StyledBadgeRight>

            </Grid>
          </Link>
      </Button>
    </ThemeProvider>
    )
}

const mapStateToProps = state =>{
  return{
    totalBasketCount: getTotalBasketCount(state),
    totalPrice: getTotalBasketPrice(state)
  }
}

export default connect(mapStateToProps, null)(BasketCart)
