// components/categories/index.js
import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router'
import { compose } from 'redux'
import classNames from 'classnames'
import * as R from 'ramda'

import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import Typography from '@material-ui/core/Typography'
import { deepOrange, brown } from '@material-ui/core/colors'
import Divider from '@material-ui/core/Divider'

import { 
  getCategories,
  getActiveCategoryId
} from 'selectors'

const Categories = ({categories, activeCategoryId})=> {
 
  const getActiveState = R.propEq('id', activeCategoryId)

  const renderCategory = (category, index)=> {
    const linkClass = classNames({
        'list-group-item': true,
        'active': getActiveState(category)
    })
    return (
      <Typography key={index} component='div'>
        <Divider />
        <ListItem button >
          <Link
                to={`/categories/${category.id}`}
                className={linkClass}
                key={index}
                style={{
                        width: '100%',
                        textDecoration: 'none',
                        fontSize: 20,
                        color: brown[800],
                        fontWeight: 'bold',
                      }}
          >
            {category.name}
          </Link>
        </ListItem>
      </Typography>
    )
  }

  const renderAllCategory = ()=> {
    const linkClass = classNames({
      'list-group-item': true,
      'active': R.isNil(activeCategoryId)
    })

    return(
      <ListItem button >
        <Link to='/' className={linkClass}
              style={{
                      width: '100%',
                      textDecoration: 'none',
                      fontSize: 20,
                      color: brown[800],
                      fontWeight: 'bold',                  
                    }}
        >
          All
        </Link>
      </ListItem>
    )
  }

  return (
    <List component="nav" aria-labelledby="brand" >
      <Typography component="h2" style={{fontWeight: "bold", fontSize: 20, color: deepOrange[900]}}>
        Brand
      </Typography>
      <List component="nav" aria-label="main folders">
        {renderAllCategory()}
        {categories.map((category, index)=> renderCategory(category, index))}
      </List>
    </List>
  )
}

const mapStateToProps = (state, ownProps) => ({
  categories: getCategories(state),
  activeCategoryId: getActiveCategoryId(ownProps)
})

export default compose(
  withRouter,
  connect(mapStateToProps, null)
)(Categories)
